plugins/SitesManager/API.php

```diff
@@ -411,13 +411,18 @@ private function getSitesFromIds($idSites, $limit = false)
 
     protected function getNormalizedUrls($url)
     {
-        if (strpos($url, 'www.') !== false) {
-            $urlBis = str_replace('www.', '', $url);
-        } else {
-            $urlBis = str_replace('://', '://www.', $url);
-        }
-
-        return array($url, $urlBis);
+        // if found, remove scheme and www. from URL
+        $hostname = str_replace('www.', '', $url);
+        $hostname = str_replace('http://', '', $hostname);
+        $hostname = str_replace('https://', '', $hostname);
+
+        // return all variations of the URL
+        return array(
+            "http://" . $hostname,
```

Comment:
>  Maybe we could always include the original $url as well` Seems a bit restrictive otherwise

Good point. Actually it does not work yet in Piwik to create websites with URLs that have weird scheme. Right now it does prepend the proper scheme;

![db urls](https://cloud.githubusercontent.com/assets/466765/8099511/e7716d50-1051-11e5-9037-43b4983ee3a7.png)

still I've added test case anyway to make sure getSitesIdFromSiteUrl works for those URLs.